core/api/src/services/mongoose/users.ts

```diff
@@ -40,6 +40,59 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const constructFilter = (userIds: UserId[], phoneCountryCodes: string[]) => {
+    const conditions: (
+      | { userId?: { $in: UserId[] } }
+      | { "phoneMetadata.countryCode": { $in: string[] } }
+    )[] = []
+
+    if (userIds.length > 0) {
+      conditions.push({ userId: { $in: userIds } })
+    }
+
+    if (phoneCountryCodes.length > 0) {
+      conditions.push({ "phoneMetadata.countryCode": { $in: phoneCountryCodes } })
+    }
+
+    return conditions.length > 0 ? { $and: conditions } : {}
+  }
+
+  const findByFilter = async function* ({
+    userIds,
+    phoneCountryCodes,
+  }: {
+    userIds: UserId[]
+    phoneCountryCodes: string[]
+  }): AsyncGenerator<UserId> | RepositoryError {
+    const query = constructFilter(userIds, phoneCountryCodes)
```

Comment:
please move inside try/catch... even if in this moment must not throw exceptions we are at service layer so we need to be extra careful to avoid pod crashes

Reply:
Changed